hazelcast/src/test/java/com/hazelcast/executor/SingleNodeTest.java

```diff
@@ -0,0 +1,346 @@
+package com.hazelcast.executor;
+
+import com.hazelcast.core.ExecutionCallback;
+import com.hazelcast.core.IExecutorService;
+import com.hazelcast.core.Member;
+import com.hazelcast.core.MemberSelector;
+import com.hazelcast.monitor.LocalExecutorStats;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class SingleNodeTest extends ExecutorServiceTestSupport {
+
+    private IExecutorService executor;
+
+    @Before
+    public void setUp() {
+        executor = createSingleNodeExecutorService("test", 1);
+    }
+
+    @Test
+    public void hazelcastInstanceAwareAndLocal() throws Exception {
+        HazelcastInstanceAwareRunnable task = new HazelcastInstanceAwareRunnable();
+        try {
+            executor.submit(task).get();
+        } catch (ExecutionException e) {
+            throw (Error)e.getCause();
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void submitNullTask() throws Exception {
```

Comment:
Perhaps best to have one style in the method naming. Some methods begin with test and some do not. I would try to keep this in a single test file in the same style. 

I would call it 
submit_whenNullTask

Because often you get a lot of scenario's and using this approach is becomes easier to deal with them:

submit_whenShutdown
submit_whenQueueFull
submit_when...